app/uk/gov/hmrc/helptosavereminder/controllers/EmailCallbackController.scala

```diff
@@ -41,52 +41,57 @@ class EmailCallbackController @Inject()(
 
   def handleCallBack(callBackReference: String): Action[AnyContent] = Action.async { implicit request =>
     {
-      request.body.asJson.get
+      request.body.asJson
+        .getOrElse(Json.toJson(""))
         .validate[EventsMap]
         .fold(
           { error =>
             Logger.error(s"Unable to parse Events List for CallBackRequest = $callBackReference")
+            Future.failed(new Exception("Error parsing the hts schedule"))
           }, { (eventsMap: EventsMap) =>
-            if (eventsMap.events.exists(x => (x.event == "PermanentBounce"))) {
+            if (eventsMap.events.exists(x => (x.event.contains("PermanentBounce")))) {
```

Comment:
No!  The wart is that using == is not typesafe, instead you have replaced the test with one that is type safe, but is weaker - it will pass any string containing "PermanentBounce".

As we are not using the cats libraries or whatever here, see https://www.wartremover.org/doc/warts.html which suggest a useful implementation of ===

Reply:
Yeah, I have neglected that. I will introduce '===' using cats

Reply:
Done